grace/models/datasets.py

```diff
@@ -8,40 +8,54 @@
 from grace.base import GraphAttrs, Annotation
 
 
-def dataset_from_graph(
-    graph: nx.Graph, *, n_hop: int = 1
+def dataset_from_subgraphs(
```

Comment:
Can you remind me why the distinction was required of separate functions for subgraph and whole graph?